### PR TITLE
accuracy op

### PR DESCRIPTION
**Pull request by yinhaofeng**
### PR types

Others
### PR changes

OPs
### Describe

add accuracy op
image
af1203b0cfed71cb9b519a85e03dc9d

### CODE REVIEW

File: paddle/fluid/operators/metrics/accuracy_op_npu.cc [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+namespace paddle {
+namespace operators {
+
+template <typename DeviceContext, typename T>
+class AccuracyNPUKernel : public framework::OpKernel<T> {
+ public:
+  void Compute(const framework::ExecutionContext& ctx) const override {
+    auto* pred = ctx.Input<Tensor>("Out");
+    auto* label = ctx.Input<Tensor>("Label");
+    // auto* logits = ctx.Input<Tensor>("Indices");
```
Comment (zhiqiu) [inline paddle/fluid/operators/metrics/accuracy_op_npu.cc:29]: Why Indices ignored?

File: paddle/fluid/operators/metrics/accuracy_op_npu.cc [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+    auto stream =
+        ctx.template device_context<paddle::platform::NPUDeviceContext>()
+            .stream();
+
+    // cast pred
+    Tensor tmp_pred(pred->type());
+    tmp_pred.Resize(pred->dims());
+    tmp_pred.mutable_data<int>(ctx.GetPlace());
+    auto runner_cast_pred = NpuOpRunner("Cast", {*pred}, {tmp_pred},
+                                        {{"dst_type", static_cast<int>(3)}});
```
Comment (zhiqiu) [inline paddle/fluid/operators/metrics/accuracy_op_npu.cc]: Do not use majic number 3, use the macro, like ACL_INT32.

Comment (zhiqiu) [reply paddle/fluid/operators/metrics/accuracy_op_npu.cc]: And, why cast to ACL_INT32?

File: python/paddle/fluid/tests/unittests/npu/test_accuracy_op_npu.py [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+class TestAccurary(OpTest):
+    def setUp(self):
+        self.op_type = "accuracy"
+        self.set_npu()
+        self.init_dtype()
+        np.random.seed(SEED)
+        pred = np.random.uniform(1, 2, [11, 1]).astype(self.dtype)
+        label = pred.copy()
+        accuracy = np.array([1]).astype(self.dtype)
+        correct = np.array([11 * 1]).astype(self.dtype)
```
Comment (zhiqiu) [inline python/paddle/fluid/tests/unittests/npu/test_accuracy_op_npu.py:41]: Better add more cases,  all 1 and all 0 is not enough.

File: paddle/fluid/operators/metrics/accuracy_op_npu.cc (hunk 1 above)
Comment (yinhaofeng) [reply paddle/fluid/operators/metrics/accuracy_op_npu.cc:29]: accuracy作为api，包含topk和accuracy两个op，其中我的accuracy作为op不需要使用这个输入

File: paddle/fluid/operators/metrics/accuracy_op_npu.cc (hunk 2 above)
Comment (yinhaofeng) [reply paddle/fluid/operators/metrics/accuracy_op_npu.cc]: 1，已修复为ACL_INT32和ACL_FLOAT。
2，我的accuracy作为op，接收的是topk的输出，正常来说输入都是int型，加不加cast不影响。但是输入为float型时需要转换为int型计算

File: python/paddle/fluid/tests/unittests/npu/test_accuracy_op_npu.py (hunk 3 above)
Comment (yinhaofeng) [reply python/paddle/fluid/tests/unittests/npu/test_accuracy_op_npu.py:41]: 已添加为0.5的单测，以及int型的单测

Comment (yinhaofeng) [reply python/paddle/fluid/tests/unittests/npu/test_accuracy_op_npu.py:41]: image

**Review by zhiqiu — APPROVED** [review]
